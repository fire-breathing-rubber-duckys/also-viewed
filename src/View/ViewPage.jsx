import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Item from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <PageWrapper>
      {data.map((item) => {
        if (valid.includes(item.productId)) {
          return <Item item={item} key={item.productId} />;
        }
        return undefined;
      })}
    </PageWrapper>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
