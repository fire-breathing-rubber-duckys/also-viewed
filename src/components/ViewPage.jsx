import React from 'react';
import styled from 'styled-components';
import { CSSTransitionGroup } from 'react-transition-group';
import PropTypes from 'prop-types';
import ViewItem from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <CSSTransitionGroup
      transitionName="carousel"
      transitionEnterTimeout={500}
      transitionLeaveTimeout={500}
      component="div"
    >
      <PageWrapper>
        {data.map((item) => {
          if (valid.includes(item.productId)) {
            return <ViewItem item={item} key={item.productId} />;
          }
          return undefined;
        })}
      </PageWrapper>
    </CSSTransitionGroup>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
