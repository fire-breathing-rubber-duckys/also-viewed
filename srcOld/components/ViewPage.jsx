import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ViewItem from './ViewItem';

const PageWrapper = styled.div`
  display: flex;
`;

const ViewPage = ({ data, valid }) => (
  <>
    <PageWrapper>
      {data.map((item) => {
        if (valid.includes(item.productId)) {
          return <ViewItem item={item} key={item.productId} />;
        }
        return undefined;
      })}
    </PageWrapper>
  </>
);

ViewPage.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  valid: PropTypes.instanceOf(Array).isRequired,
};

export default ViewPage;
