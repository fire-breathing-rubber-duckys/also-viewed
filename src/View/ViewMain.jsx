import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { CSSTransitionGroup } from 'react-transition-group';
import PropTypes from 'prop-types';
import ViewPage from './ViewPage';

const MainContainer = styled.div`
  display: flex;
  align-items: center;
`;

const ButtonStyleShow = styled.div`
  border-radius: 50%;
  height: 40px;
  bottom: initial;
  background: #f9f8f6;
  box-shadow: 0 0.4rem 0.4rem 0 rgba(12,11,8,.2);
  border: none;
  outline: none;
`;

const ButtonStyleHide = styled.div`
  height: 40px;
  opacity: 0;
  border: none;
  outline: none;
`;

const ViewMain = ({ data }) => {
  const [page, setPage] = useState(0);
  const [valid, setValid] = useState([1, 2, 3, 4, 5, 6]);

  useEffect(() => {
    if (page === 0) {
      setValid([1, 2, 3, 4, 5, 6]);
    } else {
      setValid([6, 7, 8, 9, 10, 11]);
    }
  }, [page]);

  let rightButton;
  let leftButton;

  if (page === 0) {
    rightButton = (
      <ButtonStyleShow>
        <button
          type="button"
          onClick={() => {
            setPage(1);
          }}
        >
          Right
        </button>
      </ButtonStyleShow>
    );
    leftButton = (
      <ButtonStyleHide>
        <button type="button"> </button>
      </ButtonStyleHide>
    );
  } else {
    leftButton = (
      <ButtonStyleShow>
        <button
          type="button"
          onClick={() => {
            setPage(0);
          }}
        >
          Left
        </button>
      </ButtonStyleShow>
    );
    rightButton = (
      <ButtonStyleHide>
        <button type="button"> </button>
      </ButtonStyleHide>
    );
  }

  return (
    <div className="main">
      <MainContainer>
        {leftButton}
        <CSSTransitionGroup
          transitionEnterTimeout={1000}
          transitionLeaveTimeout={1000}
          transitionName="carousel"
        >
          <ViewPage data={data} valid={valid} />
        </CSSTransitionGroup>
        {rightButton}
      </MainContainer>
    </div>
  );
};

ViewMain.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
};

export default ViewMain;
