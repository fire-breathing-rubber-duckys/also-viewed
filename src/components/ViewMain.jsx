import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { CSSTransitionGroup } from 'react-transition-group';
import PropTypes from 'prop-types';
import ViewPage from './ViewPage';

const MainContainer = styled.div`
  display: flex;
  align-items: center;
`;

const ViewMain = ({ data }) => {
  const [page, setPage] = useState(0);
  const [valid, setValid] = useState([1, 2, 3, 4, 5, 6]);
  const [hover, setHover] = useState(false);

  useEffect(() => {
    if (page === 0) {
      setValid([1, 2, 3, 4, 5, 6]);
      setHover(false);
    } else {
      setValid([6, 7, 8, 9, 10, 11]);
      setHover(false);
    }
  }, [page]);

  const hoverOn = () => {
    setHover(true);
  };

  const hoverOff = () => {
    setHover(false);
  };

  let rightButton;
  let leftButton;

  if (page === 0 && hover) {
    rightButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="button-hover"
        type="button"
        onClick={() => {
          setPage(1);
        }}
      >
        <i className="fa fa-long-arrow-right" />
      </button>
    );
    leftButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="hidden"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-left" />
      </button>
    );
  } else if (page === 0 && !hover) {
    rightButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="btn-right"
        type="button"
        onClick={() => {
          setPage(1);
        }}
      >
        <i className="fa fa-long-arrow-right" />
      </button>
    );
    leftButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="hidden"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-left" />
      </button>
    );
  } else if (page === 1 && hover) {
    leftButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="button-hover"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-left" />
      </button>
    );
    rightButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="hidden"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-right" />
      </button>
    );
  } else {
    leftButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="btn-left"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-left" />
      </button>
    );
    rightButton = (
      <button
        onMouseEnter={hoverOn}
        onMouseLeave={hoverOff}
        className="hidden"
        type="button"
        onClick={() => {
          setPage(0);
        }}
      >
        <i className="fa fa-long-arrow-right" />
      </button>
    );
  }

  return (
    <>
      <MainContainer>
        {leftButton}
        <CSSTransitionGroup
          transition="carousel"
          transitionEnterTimeout={500}
          transitionLeaveTimeout={500}
        >
          <ViewPage data={data} valid={valid} />
        </CSSTransitionGroup>
        {rightButton}
      </MainContainer>
    </>
  );
};

ViewMain.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
};

export default ViewMain;
