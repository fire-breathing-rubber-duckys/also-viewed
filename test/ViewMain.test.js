import React from 'react';
import { shallow } from 'enzyme';
import { expect } from 'chai';
import ViewMain from '../src/components/ViewMain';
import ViewPage from '../src/components/ViewPage';

describe('View Main Rendering', () => {
  it('should render ViewPage once', () => {
    const wrapper = shallow(<ViewMain />);
    expect(wrapper.find(ViewPage)).to.be.lengthOf(1);
  });
  it('should render one button at a time', () => {
    const wrapper = shallow(<ViewMain />);
    expect(wrapper.find('.btn')).to.be.lengthOf(1);
  })
});