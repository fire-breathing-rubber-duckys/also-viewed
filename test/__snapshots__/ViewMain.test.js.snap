// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="sc-dlfnbm CZBcd"
>
  <button
    className="hidden"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    <i
      className="fa fa-long-arrow-left"
    />
  </button>
  <span
    transition="carousel"
  >
    <div
      className="sc-gsTCUz ywWHk"
    >
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/148949?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Manduka
            </div>
            Lean Cork
            <br />
            Block
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (2)
            </label>
          </div>
          <label
            className="item-price"
          >
            $16.00
          </label>
        </div>
      </div>
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/182895?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Manduka
            </div>
            Rectangular
            <br />
            Bolster
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (2)
            </label>
          </div>
          <label
            className="item-price"
          >
            $68.00
          </label>
        </div>
      </div>
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/173243?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Manduka
            </div>
            unBLOK Recycled
            <br />
            Foam Yoga Block
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (3)
            </label>
          </div>
          <label
            className="item-price"
          >
            $24.00
          </label>
        </div>
      </div>
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/101409?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Jade
            </div>
            Cork Yoga
            <br />
            Block
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (6)
            </label>
          </div>
          <label
            className="item-price"
          >
            $14.95-$19.95
          </label>
        </div>
      </div>
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/182894?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Manduka
            </div>
            Lean Bolster
            <br />
            
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (1)
            </label>
          </div>
          <div>
            <label
              className="item-sale-discount"
            >
               $43.95
            </label>
            <label
              className="item-sale-price"
            >
               $56.00
            </label>
          </div>
        </div>
      </div>
      <div
        className="sc-bdfBwQ fCuAeK"
      >
        <img
          alt=""
          src="https://www.rei.com/media/product/843154?size=280"
        />
        <div>
          <div
            className="item-name"
          >
            <div
              className="item-brand"
            >
              Manduka
            </div>
            Recycled Foam
            <br />
            Yoga Block
          </div>
          <div>
            <label
              className="item-rating-count"
            >
               (8)
            </label>
          </div>
          <label
            className="item-price"
          >
            $16.00
          </label>
        </div>
      </div>
    </div>
  </span>
  <button
    className="btn-right"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    <i
      className="fa fa-long-arrow-right"
    />
  </button>
</div>
`;
